### ISSUE TITLE
HTML Special Chars are not parsed correctly

### ISSUE THREAD

**Issue by marcelorodrigo** [seethroughdev/obsidian-recipe-grabber#34]
## How to reproduce
When using the latest release, provide a URL that contains content using HTML special characters.
To make it easier to test and reproduce, you can use the recipe URL below:

- https://www.tudogostoso.com.br/receita/20617-pastel-de-feira.html

### Actual output
`... O pastel de feira &eacute; um cl&aacute;ssico do final de semana ...`

### Expected output
`... O pastel de feira é um clássico do final de semana ...`

> [!NOTE]
> The configuration _Prevent escaping HTML_ was `off`. 
> Although the issue was noticeable either with `on` or `off`.

**Comment by marcelorodrigo** [comment]
Oh, it seems to be the same or closely related to what is mentioned in #15 and #32
Feel free to close this one if it's not relevant.

**Comment by seethroughdev** [comment]
This is helpful @marcelorodrigo thanks so much. And yes, I've seen similar issues. I'm going to update a PR to take a new approach to escaping, and hopefully it makes things better. Keep you updated.

**Comment by seethroughdev** [comment]
Okay, in testing this one, i did notice something else that appears to be related to this site in particular. If you look at the schema directly on their page itself: 

```
"description": "Quem foi que disse que&amp;nbsp;essa maravilha&amp;nbsp;n&amp;atilde;o pode ser preparada a&amp;iacute;, na sua pr&amp;oacute;pria casa? O pastel de feira &amp;eacute; um cl&amp;aacute;ssico do final de semana,&amp;nbsp;uma estrela entre as del&amp;iacute;cias da culin&amp;aacute;ria de rua brasileira, ainda mais se vier acompanhado de um belo cop&amp;atilde;o de caldo de cana, n&amp;atilde;o &amp;eacute; mesmo? Com a sua croc&amp;acirc;ncia irresist&amp;iacute;vel e os seus recheios suculentos, este petisco tradicional tem o poder de despertar mem&amp;oacute;rias gustativas que remetem a feiras coloridas e animadas, onde o aroma sedutor da fritura flutua pelo ar. Nesta receita, iremos desvendar o segredo por tr&amp;aacute;s da massa de pastel de feira simples e perfeitamente crocante, que faz do pastel de feira uma verdadeira institui&amp;ccedil;&amp;atilde;o gastron&amp;ocirc;mica. Prepare-se para embarcar em uma jornada saborosa que vai te transportar diretamente para o cora&amp;ccedil;&amp;atilde;o pulsante das feiras populares do Brasil.&amp;nbsp;Confira como fazer uma receita de massa de pastel simples e perfeita!&amp;nbsp;",
```

It looks like they are encoding the `&amp;` before the `&nbsp;`, so it is incorrectly merging the string as `&amp;nbsp;` which is breaking the encoder.

So that's a long way of saying, I am testing a fix in general for entities, that will hopefully be up soon. But, I think this site might have an additional bug with their schema generation, so I don't think it will perfectly for this recipe. 

Hopefully that makes sense.

**Comment by marcelorodrigo** [comment]
Good catch!

Indeed, in that case in particular it looks like a bug on the site itself while providing the schema, not on the plugin side.
I'm happy to hear you're working still on parsing entities. Thank you very much!

**Comment by seethroughdev** [comment]
No problem at all! Just released v0.21.0 which has a much improved entity decoding. Going to close this for now, but feel free to reopen if you see anything else. Thanks.
